src/Farmer/Builders/Builders.VirtualNetwork.fs

```diff
@@ -79,6 +128,22 @@ type SubnetSpecBuilder () =
     [<CustomOperation "size">]
     member _.Size(state:SubnetBuildSpec, size) =
         { state with Size = size }
+    /// Sets the network security group for subnet
+    [<CustomOperation "network_security_group">]
+    member _.NetworkSecurityGroup(state:SubnetBuildSpec, nsg:string) =
+        { state with NetworkSecurityGroup = Some (Managed (Farmer.Arm.NetworkSecurityGroup.networkSecurityGroups.resourceId (ResourceName nsg))) }
+    member _.NetworkSecurityGroup(state:SubnetBuildSpec, nsg:ResourceName) =
```

Comment:
Last time I linked to a managed resource Isaac suggested I take in the IBuilder (Or the derived IBuilder) and IArmResource (or the derived IArmResource) instead of these string values.